src/plugins/workspace/public/containers/workspace_dropdown_list/workspace_dropdown_list.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useCallback, useMemo, useEffect } from 'react';
+
+import { EuiButton, EuiComboBox, EuiComboBoxOptionOption } from '@elastic/eui';
+import useObservable from 'react-use/lib/useObservable';
+import { CoreStart, WorkspaceAttribute } from '../../../../../core/public';
+
+type WorkspaceOption = EuiComboBoxOptionOption<WorkspaceAttribute>;
+
+interface WorkspaceDropdownListProps {
+  coreStart: CoreStart;
+  onCreateWorkspace: () => void;
+  onSwitchWorkspace: (workspaceId: string) => Promise<void>;
+}
+
+function workspaceToOption(workspace: WorkspaceAttribute): WorkspaceOption {
+  return { label: workspace.name, key: workspace.id, value: workspace };
+}
+
+export function WorkspaceDropdownList(props: WorkspaceDropdownListProps) {
+  const { coreStart, onCreateWorkspace, onSwitchWorkspace } = props;
+  const workspaceList = useObservable(coreStart.workspaces.client.workspaceList$, []);
+  const currentWorkspaceId = useObservable(coreStart.workspaces.client.currentWorkspaceId$, '');
+
+  const [loading, setLoading] = useState(false);
+  const [workspaceOptions, setWorkspaceOptions] = useState([] as WorkspaceOption[]);
+
+  const currentWorkspaceOption = useMemo(() => {
```

Comment:
There is a currentWorkspace$ in core.workspaces.client that can be used to get currentWorkspace, recommend to use that.

Reply:
will create another PR to change this